module/mempool/stdmap/backdata/heropool/pool.go

```diff
@@ -0,0 +1,300 @@
+package heropool
+
+import (
+	"math/rand"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type EjectionMode string
+
+const (
+	RandomEjection = EjectionMode("random-ejection")
+	LRUEjection    = EjectionMode("lru-ejection")
+)
+
+// number of entries we try to eject before giving up and ejecting LRU
+const maximumRandomTrials = 10
+
+// EIndex is data type representing an entity index in Pool.
+type EIndex uint32
+
+// cachedEntity represents a cached entity that is maintained as a double linked list node.
+type cachedEntity struct {
+	owner  uint64
+	node   link
+	id     flow.Identifier
+	entity flow.Entity
+}
+
+type Pool struct {
+	size         uint32
+	free         state // keeps track of free slots.
+	used         state // keeps track of allocated slots to cachedEntities.
+	values       []cachedEntity
+	ejectionMode EjectionMode
+}
+
+func NewPool(limit uint32, ejectionMode EjectionMode) *Pool {
+	l := &Pool{
+		free: state{
+			head: poolIndex{index: 0},
+			tail: poolIndex{index: 0},
+		},
+		used: state{
+			head: poolIndex{index: 0},
+			tail: poolIndex{index: 0},
+		},
+		values:       make([]cachedEntity, limit),
+		ejectionMode: ejectionMode,
+	}
+
+	l.initFreeEntities()
+
+	return l
+}
+
+// initFreeEntities initializes the free double linked-list with the indices of all cached entity values.
+func (e *Pool) initFreeEntities() {
```

Comment:
Why `e *Pool` and not `p *Pool`? :-)

Reply:
Fixed in https://github.com/onflow/flow-go/pull/1778/commits/8b25b6c4bef9759e50f54576bae3d9d988e10856